src/handlers/expenditures/helpers/getUpdatedSlots.ts

```diff
@@ -8,17 +8,37 @@ export const getUpdatedExpenditureSlots = (
   expenditure: ExpenditureFragment,
   slotId: number,
   fieldsToUpdate: Partial<Omit<ExpenditureSlot, 'id' | '__typename'>>,
+  previouslyUpdatedSlots?: ExpenditureFragment['slots'],
```

Comment:
Both are needed when checking for a matching slot in previouslyUpdatedSlots - it is possible that there are previouslyUpdatedSlots, but not with a matching slotId, in which case we grab the existingSlot from the expenditure. So in this instance both the previouslyUpdatedSlots and the expenditure.slots are needed.

I have still refactored so that the first argument only accepts the slots rather than the entire expenditure for the sake of consistency between the two arguments.

Reply:
Oh sorry, I did not realise that could be the case. In this scenario, is it worth making a separate function for your purposes, since it's specific to updating slots via multicall?